src/main/java/me/karakelley/http/Main.java

```diff
@@ -30,7 +27,7 @@ public static void main(String[] args, BiConsumer<Map<String, String>, ServerCon
 
     configurationBiConsumer.accept(argsHash, serverConfiguration);
 
-    HttpServer server = new HttpServer(serverConfiguration, new ParserFactory(), new ConnectionHandler());
+    HttpServer server = new HttpServer(serverConfiguration, new ConnectionHandler(), new RequestReaderFactory());
```

Comment:
Just curious: why did you choose to use a factory here?

Reply:
I need a new instance of `HttpRequestReader` for each new connection and didn't want to go stateless with this class so I thought instead of instantiating it inside of `ConnectionHandler` I could pass something in that was responsible for creating a new `HttpRequestReader` on each connection.